tiledb/sm/query/query_condition.cc

```diff
@@ -671,46 +706,206 @@ QueryCondition::apply_clause(
       return {Status_QueryConditionError(
                   "Cannot perform query comparison; Unsupported query "
                   "conditional type on " +
-                  clause.field_name_),
+                  node.field_name_),
               nullopt};
   }
 
   return {Status::Ok(), nullopt};
 }
 
+/**
```

Comment:
I think the proper way to do this would be to create a bitmap that has a value for every cells of the cell slabs, then you run the full query condition on the bitmap. With the original cell slabs and the computed bitmap, you can even determine how big the resulting cell slab will be, and make only one allocation for the final structure, which will make the performance much better! So for example, if you have (0, 0, 10), (1, 0, 5) as an input, and cell 5 of tile 0 and cell 3 of tile 1 don't match your condition, you would end up with the following bitmap (1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1). And you could easily compute that you need to allocate 4 cell slabs and populate the final result (0, 0, 5), (0, 6, 4), (1, 0, 3), (1, 4, 1).

Reply:
resolved locally